### ISSUE TITLE
fix terminate on ctrl-c multiprocessing KeyboardInterupt

### ISSUE THREAD

**Issue by dmidlo** [dmidlo/histdata.com-tools#15]

**Comment by dmidlo** [comment]
soo.... I tried out chatGPT...  I'm stunned... I've searched the internet on more than a few occasions trying to hunt this down...  always SIGINT with process.pool...   and only unresolved issues related to ProcessPoolExecutor... but the `executor.shutdown()` ?  ooh man.  just the hint i needed.  

reading a bit on the executor, (https://docs.python.org/3/library/concurrent.futures.html#concurrent.futures.Executor.shutdown) it may be necessary to handle RuntimeErrors as well.

prompt:
`With python, how do I handle a keyboardinterupt exception inside of a processpoolexecutor?`

Response:
`To handle a KeyboardInterrupt exception inside of a ProcessPoolExecutor, you can use a try/except block to catch the exception and then call the shutdown method on the ProcessPoolExecutor object to gracefully terminate the process pool. Here is an example:`

```python
from concurrent.futures import ProcessPoolExecutor

def function_to_run_in_parallel(arg):
    # Do some work
    return result

try:
    with ProcessPoolExecutor() as executor:
        results = executor.map(function_to_run_in_parallel, args)
except KeyboardInterrupt:
    executor.shutdown()
```

`In this example, the function_to_run_in_parallel function is run in parallel using the ProcessPoolExecutor, and the try/except block catches any KeyboardInterrupt exceptions that may be raised. If a KeyboardInterrupt is caught, the shutdown method is called on the ProcessPoolExecutor object to gracefully terminate the process pool.`

**Comment by dmidlo** [comment]
`shutdown(wait=False,  cancel_futures=True)`  will be the first thing to try.

**Comment by dmidlo** [comment]
Reconciled under #179. The original ProcessPoolExecutor/managed-queue interrupt problem is superseded by the sidecar lineage through commit 76336c4: the old ProcessPool runtime is removed by #170, sidecar workers catch KeyboardInterrupt for clean worker shutdown, and #178 validated the live Temporal sidecar path with no leftover worker/server processes. Closing as satisfied/superseded by the migration lineage.
